### PR TITLE
engine: unit status use cached source status

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #7343

### What is changed and how it works?
- df-service will call `get job api` which calls `dm status`, but current impl will get source status every time, if the source is down, it won't return after 30s timeout, which will make df-service `get/list job` timeout and fails. so change `unit.Status` to use cached source status.

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: engine/executor/dm/unitholder.go [hunk 1]
```diff
@@ -243,22 +248,28 @@ func (u *unitHolderImpl) Status(ctx context.Context) interface{} {
...
 // CheckAndUpdateStatus implement UnitHolder.CheckAndUpdateStatus.
 func (u *unitHolderImpl) CheckAndUpdateStatus(ctx context.Context) {
-	u.sourceStatusMu.RLock()
-	sourceStatus := u.sourceStatus
-	u.sourceStatusMu.RUnlock()
+	sourceStatus := u.getSourceStatus()
 
 	if sourceStatus == nil || time.Since(sourceStatus.UpdateTime) > sourceStatusRefreshInterval {
-		u.Status(ctx)
+		u.updateSourceStatus(ctx)
```
Comment (lance6716) [inline engine/executor/dm/unitholder.go:272]: Still need to call Status here, because the metrics related variables are updated in Status.

Comment (D3Hunter) [reply engine/executor/dm/unitholder.go:272]: which metric?

Comment (lance6716) [reply engine/executor/dm/unitholder.go:272]: for example

```
			s.metricsProxies.Metrics.RemainingTimeGauge.Set(float64(remainingSeconds))
```

this line is contained in (*Syncer).printStatus <- (*Syncer).Status

Comment (D3Hunter) [reply engine/executor/dm/unitholder.go:272]: fixed here https://github.com/pingcap/tiflow/pull/7618/commits/c55d42df659a1026549397229f95600a2bc45937

bad idea to have a status getter to have such side effects.

**Review by okJiang — APPROVED** [review]
LGTM

**Review by lance6716 — APPROVED** [review]

**Pull request merged**